src/ConstrainedPhotometryModel.cc

```diff
@@ -76,33 +76,56 @@ ConstrainedPhotometryModel::ConstrainedPhotometryModel(CcdImageList const &ccdIm
 }
 
 unsigned ConstrainedPhotometryModel::assignIndices(std::string const &whatToFit, unsigned firstIndex) {
-    // TODO DM-8046: currently ignoring whatToFit: eventually implement configurability.
     unsigned index = firstIndex;
-    for (auto &i : _chipMap) {
-        auto mapping = i.second.get();
-        // Don't assign indices for fixed parameters.
-        if (mapping->isFixed()) continue;
-        mapping->setIndex(index);
-        index += mapping->getNpar();
+    if (whatToFit.find("Model") == std::string::npos) {
+        LOGLS_ERROR(_log, "assignIndices was called and Model is *not* in whatToFit");
```

Comment:
Good point. I've changed it to a warning and now return the input index. If the code is ever called without "Model", I believe things can continue from that state. This would be made better with some enums instead of string parsing (hence DM-8046).